### PR TITLE
BufferGeometry: Renamed uv2, uv3, uv4 to uv1, uv2, uv3

### PR DESCRIPTION
**Pull request by mrdoob**
Related issue: #25788 #25938

**Description**

Before this PR we had this setup:

| GLTF | channel | attribute |
| --- | --- | --- |
| TEXCOORD_0 | 0 |  uv  |
| TEXCOORD_1 | 1 |  uv2 | 
| TEXCOORD_2 | 2 |  uv3 |
| TEXCOORD_3 | 3 |  uv4  |

This PR renames the attributes to match GLTF ids and `material.channel`:

| GLTF | channel | attribute |
| --- | --- | --- |
| TEXCOORD_0 | 0 |  ~uv0~ uv  |
| TEXCOORD_1 | 1 |  uv1 | 
| TEXCOORD_2 | 2 |  uv2 |
| TEXCOORD_3 | 3 |  uv3  |

### CODE REVIEW

File: src/core/BufferGeometry.js [hunk 1]
```diff
@@ -37,6 +37,42 @@ class BufferGeometry extends EventDispatcher {
...
+
+					if ( this.uv1 === undefined ) {
+
+						console.warn( 'THREE.BufferGeometry: The attribute uv2 has been renamed to uv1.' );
+						this.uv1 = value;
+
+					}
+
+					delete this.uv2;
+					this.uv2 = value;
```
Comment (mrdoob) [inline src/core/BufferGeometry.js:69]: The amazing things we can do in Javascript 😁

**Comment by mrdoob** [comment]
The warnings make `webgl_geometry_csg` super slow. Maybe we don't really need the warnings...

**Comment by mrdoob** [comment]
I've updated locally all the code locally and I'm not loving this new pattern:

```diff
 		this.setAttribute( 'position', new Float32BufferAttribute( vertices, 3 ) );
 		this.setAttribute( 'normal', new Float32BufferAttribute( normals, 3 ) );
-		this.setAttribute( 'uv', new Float32BufferAttribute( uvs, 2 ) );
+		this.setAttribute( 'uv0', new Float32BufferAttribute( uvs, 2 ) );
```

I think for now I'm just going to go for this:

| GLTF | channel | attribute |
| --- | --- | --- |
| TEXCOORD_0 | 0 |  uv  |
| TEXCOORD_1 | 1 |  uv1 | 
| TEXCOORD_2 | 2 |  uv2 |
| TEXCOORD_3 | 3 |  uv3  |

**Comment by mrdoob** [comment]
Hmm, I'm having a hard time trying to predict the side effects of this code:

```js
// Backwards compatibilty

Object.defineProperties( this.attributes, {
	'uv2': {
		set: function ( value ) {

			if ( this.uv1 === undefined ) {

				console.warn( 'THREE.BufferGeometry: The attribute uv2 has been renamed to uv1.' );
				this.uv1 = value;

			}

			delete this.uv2;

		},
		configurable: true
	}
} );
```

I feel this code may break more things than fix...
It may be better to not have it for now and release a patch version when we gather more use cases.

File: src/objects/Mesh.js [hunk 2]
```diff
@@ -377,13 +377,14 @@ function checkGeometryIntersection( object, material, raycaster, ray, uv, uv2, n
...
-			_uvA.fromBufferAttribute( uv2, a );
-			_uvB.fromBufferAttribute( uv2, b );
-			_uvC.fromBufferAttribute( uv2, c );
+			_uvA.fromBufferAttribute( uv1, a );
+			_uvB.fromBufferAttribute( uv1, b );
+			_uvC.fromBufferAttribute( uv1, c );
 
-			intersection.uv2 = Triangle.getInterpolation( _intersectionPoint, _vA, _vB, _vC, _uvA, _uvB, _uvC, new Vector2() );
+			intersection.uv1 = Triangle.getInterpolation( _intersectionPoint, _vA, _vB, _vC, _uvA, _uvB, _uvC, new Vector2() );
+			intersection.uv2 = intersection.uv1; // Backwards compatibility
```
Comment (LeviPesin) [inline src/objects/Mesh.js:387]: Maybe add deprecation comment `// @deprecated, r152` here?

File: src/renderers/webgl/WebGLPrograms.js [hunk 3]
```diff
@@ -37,9 +37,9 @@ function WebGLPrograms( renderer, cubemaps, cubeuvmaps, extensions, capabilities
 
 	function getChannel( value ) {
```
[the remark this answers is not in the document]
Comment (mrdoob) [reply src/renderers/webgl/WebGLPrograms.js:38]: Now that the ids are the same the code is more readable yes.

**Pull request merged**